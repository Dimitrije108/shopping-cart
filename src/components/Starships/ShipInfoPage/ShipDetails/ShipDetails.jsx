import { useState } from "react";
import styles from "./ShipDetails.module.css";
import ItemQuantity from "../../../ItemQuantity/ItemQuantity";
import formatNumber from "../../../../formatNumber";
import { useCartContext } from "../../../../App";

export default function ShipDetails({ basic, details }) {
	const [quantity, setQuantity] = useState(1);
	const { addToCart } = useCartContext();

	function increaseQuantity() {
		if (quantity < 20) {
			setQuantity(quantity + 1);
		}
	}

	function decreaseQuantity() {
		if (quantity > 1) {
			setQuantity(quantity - 1);
		}
	}

	function changeQuantity(e) {
		const value = e.target.value;
		// Set the input to empty if user inputs the value manually
		if (value === "") {
			setQuantity("");
		}
		// Set the input value
		if (value > 0 && value <= 20) {
			setQuantity(Number(value));
		}
	}

	function resetQuantity() {
		console.log("does it trigger")
		// Reset the value back to 1 if it was left empty
		if (quantity === "") {
			setQuantity(1);
		}
	}
	// Convert all unknown, empty or n/a values into N/A for consistency
	const info = Object.fromEntries(
		Object.entries(details).map(([key, val]) => {
			if (
				val === "" || 
				val === "unknown" ||
				val === "n/a"
			) {
				return [key, "N/A"];
			} else {
				return [key, val];
			}
	}));
	// Convert unknown price into "Price on Request" to better suit the
	// eCommerce nature of the project
	const truePrice = details.cost_in_credits === "unknown" 
		? "Price on Request" 
		: details.cost_in_credits;

	const finance = Math.trunc(truePrice / 24).toString();

  return (
		<div className={styles.shipDetailsCont}>
			<div className={styles.imgCont}>
				<img src={basic.image} />
			</div>
			<div className={styles.addCol}>
				<div  className={styles.addWrapper}>
					<div className={styles.priceCont}>
						<div>{formatNumber(truePrice)}</div>
						{truePrice !== "Price on Request" && <p>credits</p>}
					</div>
					<div className={styles.financeCont}>
						<div className={styles.financeOffer}>
							<p>Finance Offer</p>
							{truePrice !== "Price on Request" && 
								<p className={styles.offerPrice}>
									{`${formatNumber(finance)}/month`}
								</p>
							}
						</div>
						<button>Get Pre-Qualified</button>
					</div>
					<ItemQuantity 
						quantity={quantity}
						increase={increaseQuantity}
						decrease={decreaseQuantity}
						change={changeQuantity}
						reset={resetQuantity}
					/>
					<button 
						className={styles.addBtn}
						onClick={() => addToCart(quantity)}
					>ADD</button>
				</div>
			</div>
			<div className={styles.descriptionCont}>
				<h1>{basic.name}</h1>
				<p>{basic.description}</p>
			</div>
			<section className={styles.tableSection}>
				<h2>Specification</h2>
				<div  className={styles.tableCont}>
					<table className={styles.infoTable}>
						<tbody>
							<tr>
								<th>Model</th>
								<td>{info.model}</td>
							</tr>
							<tr>
								<th>Starship class</th>
								<td>{info.starship_class}</td>
							</tr>
							<tr>
								<th>Manufacturer</th>
								<td>{info.manufacturer}</td>
							</tr>
							<tr>
								<th>Length</th>
								<td>{info.length}</td>
							</tr>
							<tr>
								<th>Crew</th>
								<td>{info.crew}</td>
							</tr>
						</tbody>
					</table>
					<table className={styles.infoTable}>
						<tbody>
							<tr>
								<th>Passengers</th>
								<td>{info.passengers}</td>
							</tr>
							<tr>
								<th>Max athmospheric speed</th>
								<td>{info.max_atmosphering_speed}</td>
							</tr>
							<tr>
								<th>Hyperdrive rating</th>
								<td>{info.hyperdrive_rating}</td>
							</tr>
							<tr>
								<th>MGLT</th>
								<td>{info.MGLT}</td>
							</tr>
							<tr>
								<th>Cargo capacity</th>
								<td>{info.cargo_capacity}</td>
							</tr>
							<tr>
								<th>Consumables</th>
								<td>{info.consumables}</td>
							</tr>
						</tbody>
					</table>
				</div>
			</section>
		</div>
	)
};

// TODO: add info about financial offer, like:
// *The Estimated Monthly Payment is based on a 72-month term, a 
// down payment of $32,140, Annual Percentage Rate(APR) of 9.99%, 
// estimated taxes and fees in the amount of $289,260, and no trade-in.

// TODO: below that there can be a contact the dealer form. like:
// Contact the Dealer
// Call +1(888) 848-1168
// First Name 
// Last Name 
// Email 
// Phone Number 
// I am interested in this 2024 Aston Martin DB12

// Free Email Newsletter


// Contact me with an Insurance Quote


// Contact me with a Shipping Quote

// The finest exotic cars & sports cars for sale | duPont REGISTRYCheck Availability
// By pressing this button I agree to the Terms of Use and Privacy Policy.