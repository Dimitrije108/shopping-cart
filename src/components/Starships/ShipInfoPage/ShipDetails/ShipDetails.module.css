.shipDetailsCont {
	max-width: 1100px;
	margin: 2rem 1rem;
	padding: 0.5rem;

	display: grid;
	grid-template-columns: 3fr 1fr;
	grid-template-rows: auto;
	grid-template-areas: 
		"img img"
		"description price"
		"table price";
	gap: 2rem;
}

.imgCont {
	width: 100%;
	grid-area: img;
	display: flex;
	justify-content: center;
}

.imgCont > img {
	max-height: 600px;
	max-width: 100%;
	padding: 0.5rem;

	outline: 1px solid rgb(145, 145, 145);
	box-sizing: border-box;

	display: block;
}

.addCol {
	grid-area: price;
	display: grid;
  grid-template-rows: auto;
  align-content: start;
}

.addWrapper {
	padding: 0.5rem;

	background-color: rgb(20, 20, 20);
	outline: 1px solid rgb(145, 145, 145);

	display: flex;
	flex-direction: column;
	align-items: center;

	position: relative;
}

.priceCont {
	padding: 0.8rem 0 0.6rem 0;

	display: flex;
	flex-direction: column;
	justify-content: flex-end;
	align-items: center;
}

.priceCont > div {
  font-size: 2.2rem;
	text-align: center;
}

.priceCont > p {
  margin: 0;
  padding: 0;

  font-size: 1rem;
}

.financeCont {
	width: 85%;
	padding: 1rem;
	
	border-top: 1px solid rgb(104, 104, 104);
	border-bottom: 1px solid rgb(104, 104, 104);
	text-align: center;
}

.offerPrice {
	font-size: 1.5rem;
}

.financeCont > button {
	width: 100%;
	margin-top: 1rem;
	padding: 0.6rem 1.2rem;
}

.addBtn {
	width: 100%;
	margin-top: 1rem;
	padding: 0.9rem 3rem;

	font-family: 'DINNext bold';
	font-size: 1rem;
	letter-spacing: 1px;
	color: rgb(218, 218, 218);
  background-color: transparent;
  border: 1px solid rgb(104, 104, 104);
}

.addBtn:hover {
	border: 1px solid rgb(221, 221, 221);
	background-color: rgb(0, 116, 52);
}

.descriptionCont {
	grid-area: description;
}

.descriptionCont > h1,
.descriptionCont > p {
	margin: 0;
	padding-bottom: 0.5rem;
}

.descriptionCont > h1 {
	font-family: Trench;
	font-size: 2.4rem;
	color: rgb(233, 233, 233);

	text-decoration: underline rgb(163, 163, 163);
	text-decoration-thickness: 2px;
  text-underline-offset: 3px;

	transition: all 0.2s ease-in-out;
}

.descriptionCont > h1:hover {
	text-decoration-thickness: 1px;
	text-decoration: underline rgb(221, 221, 221);
}

.descriptionCont > p {
	font-family: DINNext;
	letter-spacing: 0.5px;
}

.detailsCont > div {
	display: flex;
}

.tableCont {
	grid-area: table;

	display: flex;
	gap: 1rem;
}

.tableSection > h2 {
	margin: 0;
	padding: 0.5rem 0;

	font-family: Trench;
	font-size: 2rem;
	color: rgb(233, 233, 233);
}

.infoTable {
	min-width: 300px;
	max-width: 800px;

	background-color: rgb(20, 20, 20);
	border-collapse: collapse;

	flex: 1;
}

.infoTable > tbody > tr > th {
	font-family: 'DINNext bold';
}

.infoTable > tbody > tr > th,
.infoTable > tbody > tr > td {
	border: 1px solid rgb(145, 145, 145);
	padding: 0.5rem;
	text-align: left;
	letter-spacing: 0.5px;
}

.limitExceeded {
  width: 200px;
  padding: 6px;

  font-family: DINNext;
  font-size: 0.8rem;

  background-color: rgb(20, 20, 20);
  border: 1px solid rgb(104, 104, 104);
  cursor: auto;
  
  position: absolute;
  bottom: -45px;

  z-index: 1;
}
