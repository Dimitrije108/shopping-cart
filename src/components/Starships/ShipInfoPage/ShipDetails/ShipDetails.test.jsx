import { describe, it, expect, vi } from "vitest";
import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import { useCartContext } from "../../../../hooks/useCartContext/useCartContext";
import ShipDetails from "./ShipDetails";

vi.mock("../../../../hooks/useCartContext/useCartContext", () => ({
	useCartContext: vi.fn(() => ({
		addToCart: vi.fn(),
	})),
}));

const mockBasicData = { 
	name: "Imperial Star Destroyer", 
	image: "star-destroyer.jpg", 
	description: "Imperial capital ship description", 
};

const mockAdvData = { 
	cost_in_credits: "150000000",
	model: "	Imperial I-class Star Destroyer",
	starship_class: "Star Destroyer",
	manufacturer: "Kuat Drive Yards",
	length: "1,600",
	crew: "47,060",
	passengers: "",
	max_atmosphering_speed: "975",
	hyperdrive_rating: "	2.0",
	MGLT: "60",
	cargo_capacity: "36000000",
	consumables: "2 years",
};

const mockAdvData2 = { 
	cost_in_credits: "unknown",
};

describe("Ship Details component", () => {
	describe("Snapshot test", () => {
		it("ship details component snapshot", () => {
			const shipDetails = render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			expect(shipDetails).toMatchSnapshot();
		});
	})

	describe("Render tests", () => {
		it("render ship image with correct src and alt attributes", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const img = screen.getByRole('img');
	
			expect(img).toBeInTheDocument();
			expect(img).toHaveAttribute('src', 'star-destroyer.jpg');
			expect(img).toHaveAttribute('alt', 'Imperial Star Destroyer starship');
		});
		
		it("render 'ADD' button correctly", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const addBtn = screen.getByRole('button', { name: 'ADD' });
	
			expect(addBtn).toBeInTheDocument();
			expect(addBtn.textContent).toMatch('ADD');
		});
	
		it("render 'Get Pre-Qualified' button correctly", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const qualifiedBtn = screen.getByRole('button', { name: 'Get Pre-Qualified' });
	
			expect(qualifiedBtn).toBeInTheDocument();
			expect(qualifiedBtn.textContent).toMatch('Get Pre-Qualified');
		});

		it("render price correctly", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const price = screen.getByText('150,000,000');
	
			expect(price).toHaveTextContent('150,000,000');
		});
		
		it("render ship info table rows", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const tables = screen.getAllByRole('table');
			const firstTable = within(tables[0]).getAllByRole('row');
			const secondTable = within(tables[1]).getAllByRole('row');
			const randomEl = within(tables[0]).getByText('Model');
			const randomEl2 = within(tables[1]).getByText('N/A');
	
			expect(tables.length).toBe(2);
			expect(firstTable.length).toBe(5);
			expect(secondTable.length).toBe(6);
			expect(randomEl).toBeInTheDocument();
			expect(randomEl2).toBeInTheDocument();
		});
	})	

	describe("Conditional rendering tests", () => {
		it("display finance offer when ship price is valid", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const finance = screen.getByTestId('financeOffer');
	
			expect(finance).toBeInTheDocument();
		});
	
		it("does not display finance offer when ship price is invalid", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData2}
					/>
				</MemoryRouter>
			)
	
			const finance = screen.queryByTestId('financeOffer');
	
			expect(finance).not.toBeInTheDocument();
		});
		
		it("render 'Price on Request' when price unknown", () => {
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData2}
					/>
				</MemoryRouter>
			)
	
			const price = screen.getByText('Price on Request');
	
			expect(price).toHaveTextContent('Price on Request');
		});
	})

	describe("Logic tests", () => {
		it("increase quantity by 1 via increase button", async () => {
			const user = userEvent.setup();
	
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const increaseBtn = screen.getByRole('button', { name: "+" });
			const input = screen.getByRole('spinbutton');
	
			await user.click(increaseBtn);
	
			expect(input.value).toEqual('2');
		});
	
		it("decrease quantity by 1 via decrease button", async () => {
			const user = userEvent.setup();
	
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const decreaseBtn = screen.getByRole('button', { name: "-" });
			const input = screen.getByRole('spinbutton');
	
			await user.clear(input);
			await user.type(input, '5');
			await user.click(decreaseBtn);
	
			expect(input.value).toEqual('4');
		});
	
		it("change quantity via input changes to the correct value", async () => {
			const user = userEvent.setup();
	
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
			
			const input = screen.getByRole('spinbutton');
	
			await user.clear(input);
	
			expect(input.value).toEqual('');
	
			await user.type(input, '8');
	
			expect(input.value).toEqual('8');
		});
	
		it("reset quantity to 1 if input field is left empty", async () => {
			const user = userEvent.setup();
	
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
			
			const input = screen.getByRole('spinbutton');
	
			await user.clear(input);
	
			expect(input.value).toEqual('');
			expect(input).toHaveFocus();
	
			await user.tab();
	
			expect(input).not.toHaveFocus();
			expect(input.value).toEqual('1');
		});
	
		it("add button adds item/s to cart with correct quantity", async () => {
			const user = userEvent.setup();
	
			const mockAddToCart = vi.fn();
			useCartContext.mockReturnValue({ addToCart: mockAddToCart});
	
			render(
				<MemoryRouter>
					<ShipDetails 
						basic={mockBasicData}
						details={mockAdvData}
					/>
				</MemoryRouter>
			)
	
			const input = screen.getByRole('spinbutton');
			await user.clear(input);
			await user.type(input, '8');
			const addBtn = screen.getByRole('button', { name: "ADD" });
			await user.click(addBtn);
	
			expect(mockAddToCart).toHaveBeenCalledTimes(1);
			expect(mockAddToCart).toHaveBeenCalledWith(
				expect.objectContaining({
					quantity: 8,
				})
			);
		});
	})
});