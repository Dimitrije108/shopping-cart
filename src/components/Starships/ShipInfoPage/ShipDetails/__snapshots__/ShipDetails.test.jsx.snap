// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ship Details component > Snapshot test > ship details component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_shipDetailsCont_28124d"
        data-testid="shipDetails"
      >
        <div
          class="_imgCont_28124d"
        >
          <img
            alt="Imperial Star Destroyer starship"
            src="star-destroyer.jpg"
          />
        </div>
        <div
          class="_addCol_28124d"
        >
          <div
            class="_addWrapper_28124d"
          >
            <div
              class="_priceCont_28124d"
            >
              <div>
                150,000,000
              </div>
              <p>
                credits
              </p>
            </div>
            <div
              class="_financeCont_28124d"
            >
              <div
                class="_financeOffer_28124d"
              >
                <p>
                  Finance Offer
                </p>
                <p
                  class="_offerPrice_28124d"
                  data-testid="financeOffer"
                >
                  6,250,000/month
                </p>
              </div>
              <button>
                Get Pre-Qualified
              </button>
            </div>
            <div
              class="_cont_3c2471"
            >
              <button
                class="_leftBtn_3c2471"
              >
                -
              </button>
              <input
                class="_quantityInput_3c2471"
                max="20"
                min="1"
                type="number"
                value="1"
              />
              <button
                class="_rightBtn_3c2471"
              >
                +
              </button>
            </div>
            <button
              class="_addBtn_28124d"
            >
              ADD
            </button>
          </div>
        </div>
        <div
          class="_descriptionCont_28124d"
        >
          <h1>
            Imperial Star Destroyer
          </h1>
          <p>
            Imperial capital ship description
          </p>
        </div>
        <section
          class="_tableSection_28124d"
        >
          <h2>
            Specification
          </h2>
          <div
            class="_tableCont_28124d"
          >
            <table
              class="_infoTable_28124d"
            >
              <tbody>
                <tr>
                  <th>
                    Model
                  </th>
                  <td>
                    	Imperial I-class Star Destroyer
                  </td>
                </tr>
                <tr>
                  <th>
                    Starship class
                  </th>
                  <td>
                    Star Destroyer
                  </td>
                </tr>
                <tr>
                  <th>
                    Manufacturer
                  </th>
                  <td>
                    Kuat Drive Yards
                  </td>
                </tr>
                <tr>
                  <th>
                    Length
                  </th>
                  <td>
                    1,600
                  </td>
                </tr>
                <tr>
                  <th>
                    Crew
                  </th>
                  <td>
                    47,060
                  </td>
                </tr>
              </tbody>
            </table>
            <table
              class="_infoTable_28124d"
            >
              <tbody>
                <tr>
                  <th>
                    Passengers
                  </th>
                  <td>
                    N/A
                  </td>
                </tr>
                <tr>
                  <th>
                    Max athmospheric speed
                  </th>
                  <td>
                    975
                  </td>
                </tr>
                <tr>
                  <th>
                    Hyperdrive rating
                  </th>
                  <td>
                    	2.0
                  </td>
                </tr>
                <tr>
                  <th>
                    MGLT
                  </th>
                  <td>
                    60
                  </td>
                </tr>
                <tr>
                  <th>
                    Cargo capacity
                  </th>
                  <td>
                    36000000
                  </td>
                </tr>
                <tr>
                  <th>
                    Consumables
                  </th>
                  <td>
                    2 years
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_shipDetailsCont_28124d"
      data-testid="shipDetails"
    >
      <div
        class="_imgCont_28124d"
      >
        <img
          alt="Imperial Star Destroyer starship"
          src="star-destroyer.jpg"
        />
      </div>
      <div
        class="_addCol_28124d"
      >
        <div
          class="_addWrapper_28124d"
        >
          <div
            class="_priceCont_28124d"
          >
            <div>
              150,000,000
            </div>
            <p>
              credits
            </p>
          </div>
          <div
            class="_financeCont_28124d"
          >
            <div
              class="_financeOffer_28124d"
            >
              <p>
                Finance Offer
              </p>
              <p
                class="_offerPrice_28124d"
                data-testid="financeOffer"
              >
                6,250,000/month
              </p>
            </div>
            <button>
              Get Pre-Qualified
            </button>
          </div>
          <div
            class="_cont_3c2471"
          >
            <button
              class="_leftBtn_3c2471"
            >
              -
            </button>
            <input
              class="_quantityInput_3c2471"
              max="20"
              min="1"
              type="number"
              value="1"
            />
            <button
              class="_rightBtn_3c2471"
            >
              +
            </button>
          </div>
          <button
            class="_addBtn_28124d"
          >
            ADD
          </button>
        </div>
      </div>
      <div
        class="_descriptionCont_28124d"
      >
        <h1>
          Imperial Star Destroyer
        </h1>
        <p>
          Imperial capital ship description
        </p>
      </div>
      <section
        class="_tableSection_28124d"
      >
        <h2>
          Specification
        </h2>
        <div
          class="_tableCont_28124d"
        >
          <table
            class="_infoTable_28124d"
          >
            <tbody>
              <tr>
                <th>
                  Model
                </th>
                <td>
                  	Imperial I-class Star Destroyer
                </td>
              </tr>
              <tr>
                <th>
                  Starship class
                </th>
                <td>
                  Star Destroyer
                </td>
              </tr>
              <tr>
                <th>
                  Manufacturer
                </th>
                <td>
                  Kuat Drive Yards
                </td>
              </tr>
              <tr>
                <th>
                  Length
                </th>
                <td>
                  1,600
                </td>
              </tr>
              <tr>
                <th>
                  Crew
                </th>
                <td>
                  47,060
                </td>
              </tr>
            </tbody>
          </table>
          <table
            class="_infoTable_28124d"
          >
            <tbody>
              <tr>
                <th>
                  Passengers
                </th>
                <td>
                  N/A
                </td>
              </tr>
              <tr>
                <th>
                  Max athmospheric speed
                </th>
                <td>
                  975
                </td>
              </tr>
              <tr>
                <th>
                  Hyperdrive rating
                </th>
                <td>
                  	2.0
                </td>
              </tr>
              <tr>
                <th>
                  MGLT
                </th>
                <td>
                  60
                </td>
              </tr>
              <tr>
                <th>
                  Cargo capacity
                </th>
                <td>
                  36000000
                </td>
              </tr>
              <tr>
                <th>
                  Consumables
                </th>
                <td>
                  2 years
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Ship Details component > ship details component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_shipDetailsCont_28124d"
        data-testid="shipDetails"
      >
        <div
          class="_imgCont_28124d"
        >
          <img
            alt="Imperial Star Destroyer starship"
            src="star-destroyer.jpg"
          />
        </div>
        <div
          class="_addCol_28124d"
        >
          <div
            class="_addWrapper_28124d"
          >
            <div
              class="_priceCont_28124d"
            >
              <div>
                150,000,000
              </div>
              <p>
                credits
              </p>
            </div>
            <div
              class="_financeCont_28124d"
            >
              <div
                class="_financeOffer_28124d"
              >
                <p>
                  Finance Offer
                </p>
                <p
                  class="_offerPrice_28124d"
                  data-testid="financeOffer"
                >
                  6,250,000/month
                </p>
              </div>
              <button>
                Get Pre-Qualified
              </button>
            </div>
            <div
              class="_cont_3c2471"
            >
              <button
                class="_leftBtn_3c2471"
              >
                -
              </button>
              <input
                class="_quantityInput_3c2471"
                max="20"
                min="1"
                type="number"
                value="1"
              />
              <button
                class="_rightBtn_3c2471"
              >
                +
              </button>
            </div>
            <button
              class="_addBtn_28124d"
            >
              ADD
            </button>
          </div>
        </div>
        <div
          class="_descriptionCont_28124d"
        >
          <h1>
            Imperial Star Destroyer
          </h1>
          <p>
            Imperial capital ship description
          </p>
        </div>
        <section
          class="_tableSection_28124d"
        >
          <h2>
            Specification
          </h2>
          <div
            class="_tableCont_28124d"
          >
            <table
              class="_infoTable_28124d"
            >
              <tbody>
                <tr>
                  <th>
                    Model
                  </th>
                  <td>
                    	Imperial I-class Star Destroyer
                  </td>
                </tr>
                <tr>
                  <th>
                    Starship class
                  </th>
                  <td>
                    Star Destroyer
                  </td>
                </tr>
                <tr>
                  <th>
                    Manufacturer
                  </th>
                  <td>
                    Kuat Drive Yards
                  </td>
                </tr>
                <tr>
                  <th>
                    Length
                  </th>
                  <td>
                    1,600
                  </td>
                </tr>
                <tr>
                  <th>
                    Crew
                  </th>
                  <td>
                    47,060
                  </td>
                </tr>
              </tbody>
            </table>
            <table
              class="_infoTable_28124d"
            >
              <tbody>
                <tr>
                  <th>
                    Passengers
                  </th>
                  <td>
                    N/A
                  </td>
                </tr>
                <tr>
                  <th>
                    Max athmospheric speed
                  </th>
                  <td>
                    975
                  </td>
                </tr>
                <tr>
                  <th>
                    Hyperdrive rating
                  </th>
                  <td>
                    	2.0
                  </td>
                </tr>
                <tr>
                  <th>
                    MGLT
                  </th>
                  <td>
                    60
                  </td>
                </tr>
                <tr>
                  <th>
                    Cargo capacity
                  </th>
                  <td>
                    36000000
                  </td>
                </tr>
                <tr>
                  <th>
                    Consumables
                  </th>
                  <td>
                    2 years
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_shipDetailsCont_28124d"
      data-testid="shipDetails"
    >
      <div
        class="_imgCont_28124d"
      >
        <img
          alt="Imperial Star Destroyer starship"
          src="star-destroyer.jpg"
        />
      </div>
      <div
        class="_addCol_28124d"
      >
        <div
          class="_addWrapper_28124d"
        >
          <div
            class="_priceCont_28124d"
          >
            <div>
              150,000,000
            </div>
            <p>
              credits
            </p>
          </div>
          <div
            class="_financeCont_28124d"
          >
            <div
              class="_financeOffer_28124d"
            >
              <p>
                Finance Offer
              </p>
              <p
                class="_offerPrice_28124d"
                data-testid="financeOffer"
              >
                6,250,000/month
              </p>
            </div>
            <button>
              Get Pre-Qualified
            </button>
          </div>
          <div
            class="_cont_3c2471"
          >
            <button
              class="_leftBtn_3c2471"
            >
              -
            </button>
            <input
              class="_quantityInput_3c2471"
              max="20"
              min="1"
              type="number"
              value="1"
            />
            <button
              class="_rightBtn_3c2471"
            >
              +
            </button>
          </div>
          <button
            class="_addBtn_28124d"
          >
            ADD
          </button>
        </div>
      </div>
      <div
        class="_descriptionCont_28124d"
      >
        <h1>
          Imperial Star Destroyer
        </h1>
        <p>
          Imperial capital ship description
        </p>
      </div>
      <section
        class="_tableSection_28124d"
      >
        <h2>
          Specification
        </h2>
        <div
          class="_tableCont_28124d"
        >
          <table
            class="_infoTable_28124d"
          >
            <tbody>
              <tr>
                <th>
                  Model
                </th>
                <td>
                  	Imperial I-class Star Destroyer
                </td>
              </tr>
              <tr>
                <th>
                  Starship class
                </th>
                <td>
                  Star Destroyer
                </td>
              </tr>
              <tr>
                <th>
                  Manufacturer
                </th>
                <td>
                  Kuat Drive Yards
                </td>
              </tr>
              <tr>
                <th>
                  Length
                </th>
                <td>
                  1,600
                </td>
              </tr>
              <tr>
                <th>
                  Crew
                </th>
                <td>
                  47,060
                </td>
              </tr>
            </tbody>
          </table>
          <table
            class="_infoTable_28124d"
          >
            <tbody>
              <tr>
                <th>
                  Passengers
                </th>
                <td>
                  N/A
                </td>
              </tr>
              <tr>
                <th>
                  Max athmospheric speed
                </th>
                <td>
                  975
                </td>
              </tr>
              <tr>
                <th>
                  Hyperdrive rating
                </th>
                <td>
                  	2.0
                </td>
              </tr>
              <tr>
                <th>
                  MGLT
                </th>
                <td>
                  60
                </td>
              </tr>
              <tr>
                <th>
                  Cargo capacity
                </th>
                <td>
                  36000000
                </td>
              </tr>
              <tr>
                <th>
                  Consumables
                </th>
                <td>
                  2 years
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
